config.py

```diff
@@ -618,9 +582,62 @@ def get_config(parse_args=True, cfg_path=None, options=None):
     else:
         agentConfig['ssl_certificate'] = agentConfig['ca_certs']
 
+    if agentConfig['api_key'] == 'APIKEYHERE' and Platform.is_windows():
+        log.debug('Querying registry to get missing config options')
+        registry_conf = get_registry_conf(agentConfig)
+        agentConfig.update(registry_conf)
+        # If it's the collector, conf is updated
+        if can_write_conf and registry_conf:
+            log.debug('Updating conf file options: %s', registry_conf.keys())
+            update_conf_file(registry_conf, config_path)
+
     return agentConfig
 
 
+def get_registry_conf(agentConfig):
+    registry_conf = {}
+    try:
+        import _winreg
+        with _winreg.OpenKey(_winreg.HKEY_LOCAL_MACHINE,
+                             "SOFTWARE\\Datadog\\Datadog Agent") as reg_key:
+            for attribute in ['api_key', 'tags', 'hostname']:
+                option = _winreg.QueryValueEx(reg_key, attribute)[0]
+                if option != '':
+                    registry_conf[attribute] = option
+    except (ImportError, WindowsError) as e:
+        log.error('Unable to get config keys from Registry: %s', e)
+
+    return registry_conf
+
+
+def update_conf_file(registry_conf, config_path):
```

Comment:
Something like
```python
def update_conf_file(registry_conf, config_path):
    lines = []
    with open(config_path, 'r') as f:
        while True:
            line = f.readline()
            if not line:
                break

            for k, v in registry_conf.iteritems():
                if k in line:
                    line = '{}: {}'.format(k, v)
            lines.append(line)
    with open(config_path, 'w') as f:
        f.write(os.linesep.join(lines))
```

is easier to read. 

That being said, we should move that somewhere else.